### DISCUSSION TITLE
manager.py start problem with dateparser

### DISCUSSION THREAD

**Discussion by ayllonpablo** [swar/Swar-Chia-Plot-Manager#104] [Q&A]
Traceback (most recent call last):
  File "/home/pabloandresayllon/Swar-Chia-Plot-Manager/manager.py", line 4, in <module>
    from plotmanager.library.utilities.commands import start_manager, stop_manager, view, analyze_logs
  File "/home/pabloandresayllon/Swar-Chia-Plot-Manager/plotmanager/library/utilities/commands.py", line 12, in <module>
    from plotmanager.library.utilities.jobs import load_jobs
  File "/home/pabloandresayllon/Swar-Chia-Plot-Manager/plotmanager/library/utilities/jobs.py", line 10, in <module>
    from plotmanager.library.utilities.log import get_log_file_name
  File "/home/pabloandresayllon/Swar-Chia-Plot-Manager/plotmanager/library/utilities/log.py", line 1, in <module>
    import dateparser
ModuleNotFoundError: No module named 'dateparser'

and the config.yaml
# This is a single variable that should contain the location of your chia executable file. This is the blockchain executable.
#
# WINDOWS EXAMPLE: C:\Users\Swar\AppData\Local\chia-blockchain\app-1.1.2\resources\app.asar.unpacked\daemon\chia.exe
#   LINUX EXAMPLE: /usr/lib/chia-blockchain/resources/app.asar.unpacked/daemon/chia
#  LINUX2 EXAMPLE: /home/swar/chia-blockchain/venv/bin/chia
chia_location:
/home/pabloandresayllon/chia-blockchain/venv/bin/chia

manager:
  # These are the config settings that will only be used by the plot manager.
  #
  # check_interval: The number of seconds to wait before checking to see if a new job should start.
  #      log_level: Keep this on ERROR to only record when there are errors. Change this to INFO in order to see more
  #                 detailed logging. Warning: INFO will write a lot of information.
  check_interval: 60
  log_level: ERROR

log:
  # folder_path: This is the folder where your log files for plots will be saved.
  folder_path: /home/pabloandresayllon/chialogs

view:
  # These are the settings that will be used by the view.
  #
  #            check_interval: The number of seconds to wait before updating the view.
  #           datetime_format: The datetime format that you want displayed in the view. See here
  #                            for formatting: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes
  # include_seconds_for_phase: This dictates whether seconds are included in the phase times.
  #        include_drive_info: This dictates whether the drive information will be showed.
  #               include_cpu: This dictates whether the CPU information will be showed.
  #               include_ram: This dictates whether the RAM information will be showed.
  #        include_plot_stats: This dictates whether the plot stats will be showed.
  check_interval: 60
  datetime_format: "%Y-%m-%d %H:%M:%S"
  include_seconds_for_phase: false
  include_drive_info: true
  include_cpu: true
  include_ram: true
  include_plot_stats: true

notifications:
  # These are different settings in order to notified when the plot manager starts and when a plot has been completed.

  # DISCORD
  notify_discord: true
  discord_webhook_url: https://discord.com/api/webhooks/838761164476514365/HddWcbEHtdLArgdsMuUig_5BH-eU3JVY9fVf3ZwPzcoLhfX7s27xAQTUgqedVyAA2LpU

  # PLAY AUDIO SOUND
  notify_sound: false
  song: audio.mp3

  # PUSHOVER PUSH SERVICE
  notify_pushover: false
  pushover_user_key: xx
  pushover_api_key: xx
  
  # TWILIO
  notify_twilio: false
  twilio_account_sid: xxxxx
  twilio_auth_token: xxxxx
  twilio_from_phone: +1234657890
  twilio_to_phone: +1234657890

progress:
  # phase_line_end: These are the settings that will be used to dictate when a phase ends in the progress bar. It is
  #                 supposed to reflect the line at which the phase will end so the progress calculations can use that
  #                 information with the existing log file to calculate a progress percent.
  #   phase_weight: These are the weight to assign to each phase in the progress calculations. Typically, Phase 1 and 3
  #                 are the longest phases so they will hold more weight than the others.
  phase1_line_end: 801
  phase2_line_end: 834
  phase3_line_end: 2474
  phase4_line_end: 2620
  phase1_weight: 33.4
  phase2_weight: 20.43
  phase3_weight: 42.29
  phase4_weight: 3.88

global:
  # These are the settings that will be used globally by the plot manager.
  #
  # max_concurrent: The maximum number of plots that your system can run. The manager will not kick off more than this
  #                 number of plots total over time.
  max_concurrent: 10

jobs:
  # These are the settings that will be used by each job. Please note you can have multiple jobs and each job should be
  # in YAML format in order for it to be interpreted correctly. Almost all the values here will be passed into the
  # Chia executable file.
  #
  # Check for more tails on the Chia CLI here: https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference
  #
  # name: This is the name that you want to give to the job.
  # max_plots: This is the maximum number of jobs to make in one run of the manager. Any restarts to manager will reset
  #            this variable. It is only here to help with short term plotting.
  #
  # [OPTIONAL] farmer_public_key: Your farmer public key. If none is provided, it will not pass in this variable to the
  #                               chia executable which results in your default keys being used. This is only needed if
  #                               you have chia set up on a machine that does not have your credentials.
  # [OPTIONAL] pool_public_key: Your pool public key. Same information as the above.
  #
  # temporary_directory: Only a single directory should be passed into here. This is where the plotting will take place.
  # [OPTIONAL] temporary2_directory: Can be a single value or a list of values. This is an optional parameter to use in
  #                                  case you want to use the temporary2 directory functionality of Chia plotting.
  # destination_directory: Can be a single value or a list of values. This is the final directory where the plot will be
  #                        transferred once it is completed. If you provide a list, it will cycle through each drive
  #                        one by one.
  #
  # size: This refers to the k size of the plot. You would type in something like 32, 33, 34, 35... in here.
  # bitfield: This refers to whether you want to use bitfield or not in your plotting. Typically, you want to keep
  #           this as true.
  # threads: This is the number of threads that will be assigned to the plotter. Only phase 1 uses more than 1 thread.
  # buckets: The number of buckets to use. The default provided by Chia is 128.
  # memory_buffer: The amount of memory you want to allocate to the process.
  # max_concurrent: The maximum number of plots to have for this job at any given time.
  # max_concurrent_with_start_early: The maximum number of plots to have for this job at any given time including
  #                                  phases that started early.
  # stagger_minutes: The amount of minutes to wait before the next job can get kicked off. You can even set this to
  #                  zero if you want your plots to get kicked off immediately when the concurrent limits allow for it.
  # max_for_phase_1: The maximum number of plots on phase 1 for this job.
  # concurrency_start_early_phase: The phase in which you want to start a plot early. It is recommended to use 4 for
  #                                this field.
  # concurrency_start_early_phase_delay: The maximum number of seconds to wait before a new plot gets kicked off when
  #                                      the start early phase has been detected.
  # temporary2_destination_sync: This field will always submit the destination directory as the temporary2 directory.
  #                              These two directories will be in sync so that they will always be submitted as the
  #                              same value.
  - name: ssd2_1
    max_plots: 999
    farmer_public_key:
    pool_public_key:
    temporary_directory: mnt/ssd2
    temporary2_directory:
    destination_directory: mnt/hdd1n8
    size: 32
    bitfield: true
    threads: 4
    buckets: 128
    memory_buffer: 6000
    max_concurrent: 3
    max_concurrent_with_start_early: 4
    stagger_minutes: 90
    max_for_phase_1: 2
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 0
    temporary2_destination_sync: false

**Comment by Amneamnius** [comment]
dateparser is one of the requirements in requirements.txt, can you try re-running the step to install requirements?

**Comment by swar** [comment]
And please pose the config.yaml in a better format because this is really hard to read

**Comment by oskarwkarlsson** [comment]
Had the same issue. Python noob. I had the chia env activated so I guess I was not in the global python context, just the local chia. Just create new shell or exit chias exectution context fixed it for me.
